src/amo/components/AutoSearchInput/index.js

```diff
@@ -236,8 +225,12 @@ export class AutoSearchInputBase extends React.Component<InternalProps, State> {
       this.searchInput.blur();
     }
 
-    const filters = this.createFiltersFromQuery(this.state.searchValue);
-    this.props.onSearch(filters);
+    const { onSearch } = this.props;
+
+    if (onSearch) {
+      const filters = this.createFiltersFromQuery(this.state.searchValue);
+      onSearch(filters);
```

Comment:
I don't see test coverage for not calling this when `onSearch` is undefined. Even though we have Flow it would be nice to get a test for it.

Reply:
If it is `undefined`, how do we test it? 🤔

Reply:
An adequate test would be to trigger a search with an undefined `onSearch` prop and make sure no errors are thrown. For example, this would have thrown an error:

```diff
diff --git a/src/amo/components/AutoSearchInput/index.js b/src/amo/components/AutoSearchInput/index.js
index dc99b8af4..1ca6c95b7 100644
--- a/src/amo/components/AutoSearchInput/index.js
+++ b/src/amo/components/AutoSearchInput/index.js
@@ -227,10 +227,8 @@ export class AutoSearchInputBase extends React.Component<InternalProps, State> {
 
     const { onSearch } = this.props;
 
-    if (onSearch) {
-      const filters = this.createFiltersFromQuery(this.state.searchValue);
-      onSearch(filters);
-    }
+    const filters = this.createFiltersFromQuery(this.state.searchValue);
+    onSearch(filters);
   };
 
   handleSearchChange = (
```

However, I thought about it more. I think it's OK to just rely on type safety here. No need to add a test.